### ISSUE TITLE
SMP SML error

### ISSUE THREAD

**Issue by ri4a** [phax/phoss-smp#137]
We are getting this since 5.2.4:

```
[2020-07-03T14:03:41,471] [SMP-SERVER] [ERROR] [http-nio-8082-exec-10] [REST API] SMP SML error -- com.helger.phoss.smp.rest2.Rest2ExceptionMapper._logRestException(Rest2ExceptionMapper.java:54)
com.helger.phoss.smp.exception.SMPSMLException: Failed to create 'iso6523-actorid-upis::xxxx:xxxx' in SML
        at com.helger.phoss.smp.backend.sql.mgr.SMPServiceGroupManagerSQL.createSMPServiceGroup(SMPServiceGroupManagerSQL.java:142) ~[phoss-smp-backend-sql-5.2.4.jar:5.2.4]
        at com.helger.phoss.smp.backend.sql.mgr.SMPServiceGroupManagerSQL.createSMPServiceGroup(SMPServiceGroupManagerSQL.java:56) ~[phoss-smp-backend-sql-5.2.4.jar:5.2.4]
        at com.helger.phoss.smp.restapi.SMPServerAPI.saveServiceGroup(SMPServerAPI.java:330) ~[phoss-smp-backend-5.2.4.jar:5.2.4]
        at com.helger.phoss.smp.rest2.APIExecutorServiceGroupPut.invokeAPI(APIExecutorServiceGroupPut.java:85) ~[phoss-smp-webapp-5.2.4.jar:5.2.4]
        at com.helger.photon.api.InvokableAPIDescriptor.invokeAPI(InvokableAPIDescriptor.java:201) ~[ph-oton-api-8.2.5.jar:8.2.5]
        at com.helger.photon.api.APIInvoker.invoke(APIInvoker.java:74) [ph-oton-api-8.2.5.jar:8.2.5]
        at com.helger.phoss.smp.rest2.Rest2Filter.onFilterBefore(Rest2Filter.java:216) [phoss-smp-webapp-5.2.4.jar:5.2.4]
        at com.helger.xservlet.AbstractXFilterUnifiedResponse.onFilterBefore(AbstractXFilterUnifiedResponse.java:99) [ph-xservlet-9.3.0.jar:9.3.0]
        at com.helger.xservlet.AbstractXFilter.doHttpFilter(AbstractXFilter.java:185) [ph-xservlet-9.3.0.jar:9.3.0]
        at com.helger.servlet.filter.AbstractHttpServletFilter.doFilter(AbstractHttpServletFilter.java:66) [ph-servlet-9.3.0.jar:9.3.0]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
        at com.helger.servlet.filter.CharacterEncodingFilter.doHttpFilter(CharacterEncodingFilter.java:187) [ph-servlet-9.3.0.jar:9.3.0]
        at com.helger.servlet.filter.AbstractHttpServletFilter.doFilter(AbstractHttpServletFilter.java:66) [ph-servlet-9.3.0.jar:9.3.0]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202) [catalina.jar:9.0.36]
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [catalina.jar:9.0.36]
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541) [catalina.jar:9.0.36]
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) [catalina.jar:9.0.36]
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) [catalina.jar:9.0.36]
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690) [catalina.jar:9.0.36]
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) [catalina.jar:9.0.36]
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) [catalina.jar:9.0.36]
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:373) [tomcat-coyote.jar:9.0.36]
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:9.0.36]
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868) [tomcat-coyote.jar:9.0.36]
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1590) [tomcat-coyote.jar:9.0.36]
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:9.0.36]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:9.0.36]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
Caused by: com.helger.phoss.smp.smlhook.RegistrationHookException: Could not create business iso6523-actorid-upis::xxxx:xxxx in SML
        at com.helger.phoss.smp.smlhook.RegistrationHookWriteToSML.createServiceGroup(RegistrationHookWriteToSML.java:140) ~[phoss-smp-backend-5.2.4.jar:5.2.4]
        at com.helger.phoss.smp.backend.sql.mgr.SMPServiceGroupManagerSQL.lambda$createSMPServiceGroup$5fd54f69$1(SMPServiceGroupManagerSQL.java:106) ~[phoss-smp-backend-sql-5.2.4.jar:5.2.4]
        at com.helger.db.jpa.JPAEnabledManager.lambda$doInTransaction$3(JPAEnabledManager.java:290) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.db.jpa.JPAEnabledManager.doInTransaction(JPAEnabledManager.java:330) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.db.jpa.JPAEnabledManager.doInTransaction(JPAEnabledManager.java:289) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.db.jpa.JPAEnabledManager.doInTransaction(JPAEnabledManager.java:303) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.phoss.smp.backend.sql.mgr.SMPServiceGroupManagerSQL.createSMPServiceGroup(SMPServiceGroupManagerSQL.java:90) ~[phoss-smp-backend-sql-5.2.4.jar:5.2.4]
        ... 32 more
Caused by: com.sun.xml.ws.fault.ServerSOAPFaultException: Client received SOAP Fault from server: The given SOAPAction http://busdox.org/serviceMetadata/ManageBusinessIdentifierService/1.0/        :createIn does not match an operation. Please see the server log to find more detail regarding exact cause of the failure.
        at com.sun.xml.ws.fault.SOAP11Fault.getProtocolException(SOAP11Fault.java:193) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.xml.ws.fault.SOAPFaultBuilder.createException(SOAPFaultBuilder.java:131) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.xml.ws.client.sei.StubHandler.readResponse(StubHandler.java:253) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.xml.ws.db.DatabindingImpl.deserializeResponse(DatabindingImpl.java:203) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.xml.ws.db.DatabindingImpl.deserializeResponse(DatabindingImpl.java:290) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.xml.ws.client.sei.SyncMethodHandler.invoke(SyncMethodHandler.java:119) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.xml.ws.client.sei.SyncMethodHandler.invoke(SyncMethodHandler.java:92) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.xml.ws.client.sei.SEIStub.invoke(SEIStub.java:161) ~[jaxws-rt-2.2.10.jar:2.2.10]
        at com.sun.proxy.$Proxy105.create(Unknown Source) ~[?:?]
        at com.helger.peppol.smlclient.ManageParticipantIdentifierServiceCaller.create(ManageParticipantIdentifierServiceCaller.java:172) ~[peppol-sml-client-8.1.1.jar:8.1.1]
        at com.helger.peppol.smlclient.ManageParticipantIdentifierServiceCaller.create(ManageParticipantIdentifierServiceCaller.java:137) ~[peppol-sml-client-8.1.1.jar:8.1.1]
        at com.helger.phoss.smp.smlhook.RegistrationHookWriteToSML.createServiceGroup(RegistrationHookWriteToSML.java:128) ~[phoss-smp-backend-5.2.4.jar:5.2.4]
        at com.helger.phoss.smp.backend.sql.mgr.SMPServiceGroupManagerSQL.lambda$createSMPServiceGroup$5fd54f69$1(SMPServiceGroupManagerSQL.java:106) ~[phoss-smp-backend-sql-5.2.4.jar:5.2.4]
        at com.helger.db.jpa.JPAEnabledManager.lambda$doInTransaction$3(JPAEnabledManager.java:290) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.db.jpa.JPAEnabledManager.doInTransaction(JPAEnabledManager.java:330) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.db.jpa.JPAEnabledManager.doInTransaction(JPAEnabledManager.java:289) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.db.jpa.JPAEnabledManager.doInTransaction(JPAEnabledManager.java:303) ~[ph-db-jpa-6.2.0.jar:6.2.0]
        at com.helger.phoss.smp.backend.sql.mgr.SMPServiceGroupManagerSQL.createSMPServiceGroup(SMPServiceGroupManagerSQL.java:90) ~[phoss-smp-backend-sql-5.2.4.jar:5.2.4]
        ... 32 more
```

Deleting is similarly broken.

**Comment by phax** [comment]
I heard similar rumors previously, and sometimes it magically works.
Do you have that issue on test or production?

**Comment by ri4a** [comment]
Both. But note I upgraded from 5.2.1 and downgrading to 5.2.3 did not resolve it, so I think it was introduced in 5.2.2.

**Comment by phax** [comment]
Or it is a bug on CEF side, that they did an incompatible update. Others are running 5.2.3 without problems. I file an issue on their side.

**Comment by phax** [comment]
v5.2.3 was out for 4 month - if no registration worked, I guess somebody would have contacted me.... (hopefully)

**Comment by JonasZal** [comment]
> v5.2.3 was out for 4 month - if no registration worked, I guess somebody would have contacted me.... (hopefully)

I can assure you that this version works, at least in the production environment.
I use SQL edition.

**Comment by phax** [comment]
Initial answer from CEF:
> There was not change on production or acceptance.
> We also did a test now on production (also the service with spaces in actions) and everything worked as it should.
> If you want it to be investigated, please send more data (when this error happened and for which user).

**Comment by phax** [comment]
@ri4a  - this is an error in 5.2.4
Please downgrade to 5.2.3

**Comment by phax** [comment]
Fixed in 5.2.5
